### PR TITLE
Removed all code related to sign in through google from app.

### LINKED ISSUE #3314: Remove code related to google sign in
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.

### What would you like to improve?

- remove code related to google sign in as we are not using it anymore

### Because...

- makes code cleaner

### Description

Couple places which I think can be removed
- `loginWithGoogle` method in `OnboardingViewModel`
- `googleSignInLauncher` method in `RootActivity`

### Success Criteria

_No response_

### PR DESCRIPTION
**Pull request by ahsanishaq721**
…ts and files.

## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [ ] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?
Describe with a few bullets **what's new:**

- ...

- ...

## Risk factors

**What may go wrong if we merge your PR?**

- ...
- ...

**In what cases won't your code work?**

- ...
- ...

## Does this PR close any GitHub issues?

- Closes #3314

## Troubleshooting CI failures ❌

Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Review by ILIYANGERMANOV — APPROVED** [review]
@ahsanishaq721 fix merge conflicts and add "Closes #ISSUE_NUMBER" to the PR description

**Comment by ahsanishaq721** [comment]
Added close in PR description but not understanding the merging conflicts. Can you help? What i have to do

**Comment by ILIYANGERMANOV** [comment]
Screenshot_20240711_203957_Chrome

click "Resolve Conflicts" and make sure that both Google dependencies are removed

**Comment by ahsanishaq721** [comment]
@ILIYANGERMANOV done with resolved conflicts.

**Comment by ILIYANGERMANOV** [comment]
> @ILIYANGERMANOV done with resolved conflicts.

@ahsanishaq721 now you need to fix the PR checks. Detekt is failing. Use the guide from the description

**Comment by suyash01** [comment]
@ahsanishaq721 run gradle task for detekt in local and fix the issues until build is successful. For windows you can run `.\gradlew detekt` in the root of your project folder.

**Comment by ahsanishaq721** [comment]
Thank u @suyash01

**Comment by ahsanishaq721** [comment]
@ILIYANGERMANOV  I am facing gradle issues, can I do some fixes in gradle in same branch and should i make another PR?

**Comment by ahsanishaq721** [comment]
issue
@ILIYANGERMANOV facing this issue while runing detekt command. 
My task was to remove google sign in related code which i have done. This is related to existing code. can you help me with that?

**Comment by suyash01** [comment]
> issue
> @ILIYANGERMANOV facing this issue while runing detekt command. 
> My task was to remove google sign in related code which i have done. This is related to existing code. can you help me with that? 

There should be detekt lint error message above the build failed error message. It is not highlighted red.

**Comment by ahsanishaq721** [comment]
@suyash01 what should I do now? now detekt will be passed?

**Comment by ILIYANGERMANOV** [comment]
@ahsanishaq721 read the CI Troubleshooting guide. Click "Details" on the failed detekt check and you'll see

Property 'style>OptionalWhenBraces' is deprecated. Same functionality is implemented in BracesOnWhenStatements.
/home/runner/work/ivy-wallet/ivy-wallet/app/src/main/java/com/ivy/wallet/RootActivity.kt:207:1: First line in a method block should not be empty [NoEmptyFirstLineInMethodBlock]
/home/runner/work/ivy-wallet/ivy-wallet/app/src/main/java/com/ivy/wallet/RootActivity.kt:29:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/app/src/main/java/com/ivy/wallet/RootActivity.kt:30:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/app/src/main/java/com/ivy/wallet/RootActivity.kt:55:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/screen/onboarding/src/main/java/com/ivy/onboarding/viewmodel/OnboardingViewModel.kt:25:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/screen/onboarding/src/main/java/com/ivy/onboarding/viewmodel/OnboardingViewModel.kt:43:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/app/src/main/java/com/ivy/wallet/RootActivity.kt:71:26: Private property `onGoogleSignInIdTokenResult` is unused. [UnusedPrivateProperty]
/home/runner/work/ivy-wallet/ivy-wallet/screen/onboarding/src/main/java/com/ivy/onboarding/viewmodel/OnboardingViewModel.kt:49:17: Private property `ivyContext` is unused. [UnusedPrivateProperty]

**Comment by ILIYANGERMANOV** [comment]
There are unused imports and fields

**Comment by ahsanishaq721** [comment]
Ok, but i didn't import or add anything new in code. I have just removed the google sign in related code, which was my task.

**Comment by ahsanishaq721** [comment]
@ILIYANGERMANOV creating PR again

**Comment by ahsanishaq721** [comment]
@ILIYANGERMANOV I have done previously mentioned tasks mentioned by detekt. Now it is showing a longer list. These are existing issues and not related to my task I think.

**Comment by ILIYANGERMANOV** [comment]
> @ILIYANGERMANOV I have done previously mentioned tasks mentioned by detekt. Now it is showing a longer list. These are existing issues and not related to my task I think.

You can either:
- fix them
- use Suppress for the harder
- or add a Detekt baseline 

Read the Troubleshooting guide

**Comment by ahsanishaq721** [comment]
i have run ./gradlew detektBaseline command but it failed the build process.

**Comment by ILIYANGERMANOV** [comment]
@ahsanishaq721 can you trying using `@Suppress` for the hard fixes and fix the easier one? It shouldn't be that hard. Last week, I fixed like 400 or so 😅

**Comment by ahsanishaq721** [comment]
@ILIYANGERMANOV At last all checks have been passed. I was first time working on detekt so sorry for delay.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -104,7 +104,7 @@ hilt-gradle-plugin = { module = "com.google.dagger:hilt-android-gradle-plugin",
 
 # Google
 google-services-plugin = { module = "com.google.gms:google-services", version = "4.4.2" }
-google-playservices-auth = { module = "com.google.android.gms:play-services-auth", version = "21.2.0" }
+google-play-asset = { module = "com.google.android.play:asset-delivery", version = "2.2.2" }
```
Comment (ILIYANGERMANOV) [inline gradle/libs.versions.toml:107]: This library must be removed. That's caused by an incorrect fix of the merge conflict

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -209,7 +209,7 @@ glance = [
     "glance-material3"
 ]
 google = [
-    "google-playservices-auth",
+    "google-play-asset",
```
Comment (ILIYANGERMANOV) [inline gradle/libs.versions.toml:212]: Remove both

File: app/build.gradle.kts [hunk 3]
```diff
@@ -8,6 +8,7 @@ plugins {
     id("com.google.devtools.ksp")
     id("com.google.gms.google-services")
     id("com.google.firebase.crashlytics")
+    id("io.gitlab.arturbosch.detekt")
```
Comment (ILIYANGERMANOV) [inline app/build.gradle.kts:11]: Why do we need this?

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Nice job fixing the CI 👏 Found some issues that we need to fix before merging

File: config/detekt/config.yml [hunk 4]
```diff
@@ -513,8 +513,6 @@ style:
   # Use it now because BracesOnWhenStatements doesn't work as expected
   # https://github.com/detekt/detekt/issues/6377
   # Delete OptionalWhenBraces rule config after detekt#6377 fix
-  OptionalWhenBraces:
-    active: false
```
Comment (ILIYANGERMANOV) [inline config/detekt/config.yml:517]: Remove the comment above

File: app/build.gradle.kts (hunk 3 above)
Comment (ahsanishaq721) [reply app/build.gradle.kts:11]: There was an error that your app folder didn't have detekt

Comment (ILIYANGERMANOV) [reply app/build.gradle.kts:11]: Got it, that's fine. Let's keep it

File: gradle/libs.versions.toml (hunk 1 above)
Comment (ILIYANGERMANOV) [reply gradle/libs.versions.toml:107]: Because it was removed in another PR before yours that's already been merged

Comment (ILIYANGERMANOV) [reply gradle/libs.versions.toml:107]: https://github.com/Ivy-Apps/ivy-wallet/pull/3330

File: config/detekt/config.yml (hunk 4 above)
Comment (ahsanishaq721) [reply config/detekt/config.yml:517]: ok

**Review by ILIYANGERMANOV — APPROVED** [review]
LGTM!

**Comment by ahsanishaq721** [comment]
Thank u. Nice to work on this project.  This was my first contribution to any open source project.

**Comment by ahsanishaq721** [comment]
Now, lint have failed. It happened for the first time. Why?

**Comment by ILIYANGERMANOV** [comment]
> Now, lint have failed. It happened for the first time. Why?

Retrying it, seems like a GitHub Actions problem

**Comment by ahsanishaq721** [comment]
Ok done. Nice to work with you. It was my first small contribution to any open source project.

**Pull request merged**